### PR TITLE
Add support for Micronaut annotations.

### PR DESCRIPTION
**Pull request by kreiger**
Hi! This adds support for Micronaut annotations: https://docs.micronaut.io/latest/api/io/micronaut/http/annotation/package-summary.html

This is quick and dirty with duplication of existing classes. so should probably be refactored for less duplication.

### CODE REVIEW

**Comment by viartemev** [comment]
@kreiger thanks a lot for contributing.
1. Do you think about `@Versioning` annotation? It can be a different feature and can be done in another PR.
2. I've tested this PR and found that path 
```
@Get("/{id:4}")
public Optional<Person> findById(@NotNull Integer id)
```
doesn't handle properly. You can't find it by typing, for example `/4` or `/12313`. 

The same issue for:
```
@Get("{?max,offset}")
public List<Person> findAll(@Nullable Integer max, @Nullable Integer offset)
```

You can test it by yourself on this sample project: https://github.com/piomin/sample-micronaut-applications

If you need any help with debugging/running plugin, let me know

**Comment by kreiger** [comment]
Thanks for your comments! I will have a look and see if i can fix the issues you mention. :)

File: src/main/kotlin/com/viartemev/requestmapper/annotations/micronaut/MicronautMappingAnnotation.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                val expression = extractExpression(pathVariableValue)
+                if (expression.isNotBlank()) expression else defaultValue
+            }
+            else -> defaultValue
+        }
+    }
+
+    companion object {
+        private const val CONTROLLER_ANNOTATION = "io.micronaut.http.annotation.Controller"
+        private const val CLIENT_ANNOTATION = "io.micronaut.http.client.annotation.Client"
```
[the remark this answers is not in the document]
Comment (kreiger) [reply src/main/kotlin/com/viartemev/requestmapper/annotations/micronaut/MicronautMappingAnnotation.kt]: Not really no, as i understand your question.

The `@Client` annotation is used to generate a http client for the corresponding controller.
I find it very useful to be able to find both the generated clients as well as the controllers.

**Comment by kreiger** [comment]
BTW, i assume you mean `io.micronaut.core.version.annotation.Version`?

I had not thought about it since we don't use it in any project (yet).

File: src/main/kotlin/com/viartemev/requestmapper/annotations/micronaut/MicronautMappingAnnotation.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (viartemev) [reply src/main/kotlin/com/viartemev/requestmapper/annotations/micronaut/MicronautMappingAnnotation.kt]: It looks like a different functionality, which can be mapped to another shortkey.
You could raise a ticket for it, also Feign client can be used as a client.

**Comment by viartemev** [comment]
@kreiger Do you need help with the PR?

**Comment by kreiger** [comment]
Hi again! I'm sorry i didn't get back to you. Thank you so much for picking it up!

**Pull request closed without merging**